data/processors/nat.py

```diff
@@ -52,57 +57,218 @@ def merge_nat_buildings(data):
 
     b_data = data[internal_id]
     b_data["nat_data"] = building.as_dict()
 
     # NAT buildings are merged after TUMonline and the MyTUM Roomfinder. So if the others
     # weren't used as sources, but the NAT Roomfinder has this building, we know it's from there.
+    # All buildings are at least in the areatree, which is always the first source.
     base_sources = b_data.setdefault("sources", {}).setdefault("base", [])
-    if not base_sources:
+    if len(base_sources) == 1:
         base_sources.append(
             {
                 "name": "NAT Roomfinder",
                 "url": f"https://www.ph.tum.de/about/visit/roomfinder/?room={building.b_code}",
             },
         )
-    b_data.setdefault("props", {}).setdefault("ids", {}).setdefault("b_id", building.b_id)
+    b_data.setdefault("props", {}).setdefault("ids", {}).setdefault("b_id", internal_id)
 
 
-def merge_nat_rooms(_data):
+def merge_nat_rooms(data):
     """
     Merge the rooms in the NAT Roomfinder with the existing data.
     This will not overwrite the existing data, but act directly on the provided data.
     """
 
     with open("external/results/rooms_nat.json", encoding="utf-8") as file:
-        _rooms = json.load(file)
+        rooms = json.load(file)
+
+    with open("external/results/usages_tumonline.json", encoding="utf-8") as file:
+        usages = json.load(file)
+
+    not_merged_parent = 0
+    not_merged_outdated = 0
+    for nat_id, nat_data in rooms.items():
+        b_code, id_rest = nat_id.split(".", 1)
+
+        if b_code in EXCLUDED_BUILDINGS:
+            not_merged_parent += 1
+            continue
+
+        b_id = _infer_internal_id(b_code, data)
+        internal_id = b_id + "." + id_rest
+
+        if _is_room_excluded(internal_id, b_id, data):
+            not_merged_outdated += 1
+            continue
+
+        r_data = _get_room_base(internal_id, b_id, nat_data, data, usages)
+
+        r_data.setdefault("sources", {}).setdefault("base", []).append(
+            {
+                "name": "NAT Roomfinder",
+                "url": f"https://www.ph.tum.de/about/visit/roomfinder/?room={nat_id}",
+            },
+        )
+
+        _merge_nat_extra_props(r_data, nat_data)
+
+    logging.debug(
+        f"{not_merged_parent} rooms not merged because their parent buildings were not merged.",
+    )
+    logging.debug(
+        f"{not_merged_outdated} rooms not merged because their buildings "
+        f"are not exclusively from the NAT roomfinder (possibly outdated data).",
+    )
+
+
+def _is_room_excluded(internal_id, b_id, data):
+    """
+    Some rooms are excluded from merging because they are more likely outdated (removed).
+    This concerns rooms that are so far not in `data` (so not in TUMonline), but
+    from a building that is not only only from the NAT Roomfinder. In other words,
+    new rooms are only added if the entire building is only added because of the
+    NAT Roomfinder.
```

Comment:
I tink this might be a bit clearer
```suggestion
    from a building that is not only only from the NAT Roomfinder. 
    In other words, new rooms are only added if the entire building is added by the
    NAT Roomfinder's data.
```